Remove `mlflow.keras` module, drop mlflow TF estimator autologging and  logging/saving TF estimator model code, integrate logging keras model code to mlflow.log_model, Support saving and loading TF2 module (#6530)
mlflow/mlflow: docs/source/models.rst

    from keras.layers import Conv2D, MaxPooling2D, Dense, Flatten
    from keras.optimizers import SGD
    import mlflow

>     import mlflow.tensorflow

Reviewer: ```suggestion
```

    model.fit(trainX, trainY, epochs=10, batch_size=32, validation_data=(testX, testY))

    signature = infer_signature(testX, model.predict(testX))

>     mlflow.tensorflow.log_model(model=model, path="mnist_cnn", signature=signature)

Reviewer: Does `mlflow.tensorflow.log_model` have a `path` argument?

>     mlflow.tensorflow.log_model(keras_model=model, path="mnist_cnn", signature=signature)

Reviewer: ```suggestion
    mlflow.tensorflow.log_model(model=model, path="mnist_cnn", signature=signature)
```

The ``tensorflow`` model flavor allows serialized TensorFlow models in
`SavedModel format <https://www.tensorflow.org/guide/saved_model#save_and_restore_models>`_

> or Keras models in MLflow Model format

Reviewer: Mentioning both `TensorFlow models` and `Keras models` is confusing. Can we remove `Keras models`?

Reviewer: In Tensorflow 2.0, is `TensorFlow models` equivalent to `Keras model`?

> The ``tensorflow`` model flavor allows serialized TensorFlow models in
> `SavedModel format <https://www.tensorflow.org/guide/saved_model#save_and_restore_models>`_
> or Keras models in MLflow Model format
> to be logged in MLflow format via the :py:func:`mlflow.tensorflow.save_model()` and

Reviewer 2: ```suggestion
The ``tensorflow`` model flavor allows TensorFlow Core models and Keras models
to be logged in MLflow format via the :py:func:`mlflow.tensorflow.save_model()` and
```

:py:func:`mlflow.tensorflow.log_model()` methods. These methods also add the ``python_function``
flavor to the MLflow Models that they produce, allowing the models to be interpreted as generic
Python functions for inference via :py:func:`mlflow.pyfunc.load_model()`. This loaded PyFunc model
can be scored with both DataFrame input and numpy array input. Finally, you can use the
:py:func:`mlflow.tensorflow.load_model()` method to load MLflow Models with the ``tensorflow``

> flavor as TensorFlow graphs or Keras model.

Reviewer 2: ```suggestion
flavor as TensorFlow Core models or Keras models.
```
